Test TA #4 partial move success failure

Failure:
```
======================================================================
ERROR: test_ta_04_partial_move_success (Game.tests.TestTakeAction)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/Drew/Projects/Tactics/Server/Game/tests.py", line 868, in setUp
    self.helper_execute_success(self.pu_cmd)
  File "/Users/Drew/Projects/Tactics/Server/Communication/testhelper.py", line 235, in helper_execute_success
    self.assertTrue(result["Success"], "Success was not TRUE: {0}".format(result))
AssertionError: Success was not TRUE: {u'Error': u'Location X:0 Y:0 is not a valid placement location for a unit for your team.', u'Success': False, u'Request_ID': u'4'}
```

Log Snippet:
```
08/26/17 01:00:31 PM testhelper:184] DEBUG ============  Starting Test: test_ta_04_partial_move_success ============
[08/26/17 01:00:31 PM create_dat: 34]  INFO Creating Version object...
[08/26/17 01:00:31 PM create_dat: 40]  INFO Creating Ability objects...
[08/26/17 01:00:31 PM create_dat: 46]  INFO Creating Action objects...
[08/26/17 01:00:31 PM create_dat: 52]  INFO Creating Leader objects...
[08/26/17 01:00:31 PM create_dat: 62]  INFO Creating Perk objects...
[08/26/17 01:00:31 PM create_dat: 69]  INFO Creating Map objects...
[08/26/17 01:00:31 PM create_dat: 75]  INFO Creating Stat objects...
[08/26/17 01:00:31 PM create_dat: 81]  INFO Creating Terrain objects...
[08/26/17 01:00:31 PM create_dat: 88]  INFO Creating Class objects...
[08/26/17 01:00:32 PM maphelper.:114] DEBUG Movement Dictionary:
[08/26/17 01:00:32 PM maphelper.:115] DEBUG {u'R': {u'Cleric': 1, u'Flier': 1, u'Archer': 1, u'Armor': 1, u'Swordsman': 1, u'Rogue': 1, u'Horseman': 1, u'Mage': 1}, u'M': {u'Cleric': 3, u'Flier': 1, u'Archer': 4, u'Armor': 99, u'Swordsman': 4, u'Rogue': 2, u'Horseman': 99, u'Mage': 3}, u'G': {u'Cleric': 1, u'Flier': 1, u'Archer': 1, u'Armor': 1, u'Swordsman': 1, u'Rogue': 1, u'Horseman': 1, u'Mage': 1}, u'W': {u'Cleric': 99, u'Flier': 1, u'Archer': 99, u'Armor': 99, u'Swordsman': 99, u'Rogue': 99, u'Horseman': 99, u'Mage': 99}, u'F': {u'Cleric': 2, u'Flier': 1, u'Archer': 2, u'Armor': 3, u'Swordsman': 3, u'Rogue': 1, u'Horseman': 3, u'Mage': 2}}
[08/26/17 01:00:32 PM testhelper:117] DEBUG {"username": "game_user_1", "Command": "CU", "pw": "aaaaaaa", "email": "userOne@email.com"}
[08/26/17 01:00:32 PM testhelper: 74] DEBUG Chn.name=Test
[08/26/17 01:00:32 PM  router.py:102] DEBUG Parsing incoming json request: 
{"username": "game_user_1", "Command": "CU", "pw": "aaaaaaa", "email": "userOne@email.com"}
[08/26/17 01:00:32 PM  router.py:142] DEBUG Received command: CU
[08/26/17 01:00:32 PM  router.py:189] DEBUG Adding channel name to JSON: Test1
[08/26/17 01:00:32 PM  router.py:247] DEBUG Response: {'Token': '4d9574081f034331b12150e68ddbd7f1', 'Success': True}
[08/26/17 01:00:32 PM testhelper:120] DEBUG {"Token": "4d9574081f034331b12150e68ddbd7f1", "Success": true}
[08/26/17 01:00:32 PM testhelper: 74] DEBUG Chn.name=Test
[08/26/17 01:00:32 PM  router.py:102] DEBUG Parsing incoming json request: 
{"username": "game_user_1", "Command": "LGN", "pw": "aaaaaaa"}
[08/26/17 01:00:32 PM  router.py:142] DEBUG Received command: LGN
[08/26/17 01:00:32 PM  router.py:189] DEBUG Adding channel name to JSON: Test1
[08/26/17 01:00:32 PM  router.py:247] DEBUG Response: {'Token': '42f3b43c44ac4a198ad859059109fa42', 'Success': True}
[08/26/17 01:00:32 PM testhelper: 74] DEBUG Chn.name=Test
[08/26/17 01:00:32 PM  router.py:102] DEBUG Parsing incoming json request: 
{"Ability": "Extra Range", "Perks": [], "Command": "ST", "Request_ID": "0", "Units": ["Cleric", "Archer", "Flier", "Armor", "Rogue", "Horseman", "Mage", "Swordsman"], "Leader": "Sniper"}
[08/26/17 01:00:32 PM  router.py:142] DEBUG Received command: ST
[08/26/17 01:00:32 PM statichelp:120] DEBUG Loading Version data...
[08/26/17 01:00:32 PM statichelp:122] DEBUG Using latest version: 1.0
[08/26/17 01:00:32 PM statichelp:125] DEBUG Loading Ability data...
[08/26/17 01:00:32 PM statichelp:210] DEBUG There are 4 objects in ability for version 1.0.
[08/26/17 01:00:32 PM statichelp:131] DEBUG Loading Action data...
[08/26/17 01:00:32 PM statichelp:237] DEBUG There are 3 objects in action for version 1.0.
[08/26/17 01:00:32 PM statichelp:137] DEBUG Loading Class data...
[08/26/17 01:00:32 PM statichelp:275] DEBUG There are 8 objects in class for version 1.0.
[08/26/17 01:00:32 PM statichelp:298] DEBUG There are 9 objects in Unit_Stat for class Swordsman version 1.0.
[08/26/17 01:00:32 PM statichelp:305] DEBUG There are 5 objects in Terrain_Unit_Movement for class	Swordsman version 1.0.
[08/26/17 01:00:32 PM statichelp:298] DEBUG There are 9 objects in Unit_Stat for class Mage version 1.0.
[08/26/17 01:00:32 PM statichelp:305] DEBUG There are 5 objects in Terrain_Unit_Movement for class	Mage version 1.0.
[08/26/17 01:00:32 PM statichelp:298] DEBUG There are 9 objects in Unit_Stat for class Horseman version 1.0.
[08/26/17 01:00:32 PM statichelp:305] DEBUG There are 5 objects in Terrain_Unit_Movement for class	Horseman version 1.0.
[08/26/17 01:00:32 PM statichelp:298] DEBUG There are 9 objects in Unit_Stat for class Rogue version 1.0.
[08/26/17 01:00:32 PM statichelp:305] DEBUG There are 5 objects in Terrain_Unit_Movement for class	Rogue version 1.0.
[08/26/17 01:00:32 PM statichelp:298] DEBUG There are 9 objects in Unit_Stat for class Armor version 1.0.
[08/26/17 01:00:32 PM statichelp:305] DEBUG There are 5 objects in Terrain_Unit_Movement for class	Armor version 1.0.
[08/26/17 01:00:32 PM statichelp:298] DEBUG There are 9 objects in Unit_Stat for class Flier version 1.0.
[08/26/17 01:00:32 PM statichelp:305] DEBUG There are 5 objects in Terrain_Unit_Movement for class	Flier version 1.0.
[08/26/17 01:00:32 PM statichelp:298] DEBUG There are 9 objects in Unit_Stat for class Archer version 1.0.
[08/26/17 01:00:32 PM statichelp:305] DEBUG There are 5 objects in Terrain_Unit_Movement for class	Archer version 1.0.
[08/26/17 01:00:33 PM statichelp:298] DEBUG There are 9 objects in Unit_Stat for class Cleric version 1.0.
[08/26/17 01:00:33 PM statichelp:305] DEBUG There are 5 objects in Terrain_Unit_Movement for class	Cleric version 1.0.
[08/26/17 01:00:33 PM statichelp:143] DEBUG Loading Leader data...
[08/26/17 01:00:33 PM statichelp:333] DEBUG There are 5 objects in leader for version 1.0.
[08/26/17 01:00:33 PM statichelp:149] DEBUG Loading Map data...
[08/26/17 01:00:33 PM statichelp:370] DEBUG There are 2 objects in map for version 1.0.
[08/26/17 01:00:33 PM statichelp:155] DEBUG Loading Perk data...
[08/26/17 01:00:33 PM statichelp:416] DEBUG There are 4 objects in perk for version 1.0.
[08/26/17 01:00:33 PM statichelp:161] DEBUG Loading Stat data...
[08/26/17 01:00:33 PM statichelp:449] DEBUG There are 9 objects in stat for version 1.0.
[08/26/17 01:00:33 PM statichelp:167] DEBUG Loading Terrain data...
[08/26/17 01:00:33 PM statichelp:484] DEBUG There are 5 objects in terrain for version 1.0.
[08/26/17 01:00:33 PM  router.py:247] DEBUG Response: {'Success': True, 'Request_ID': u'0'}
[08/26/17 01:00:33 PM testhelper: 74] DEBUG Chn.name=Test
[08/26/17 01:00:33 PM  router.py:102] DEBUG Parsing incoming json request: 
{"Command": "FM", "Request_ID": "1"}
[08/26/17 01:00:33 PM  router.py:142] DEBUG Received command: FM
[08/26/17 01:00:33 PM  router.py:247] DEBUG Response: {'Success': True, 'Request_ID': u'1'}
[08/26/17 01:00:33 PM testhelper:117] DEBUG {"username": "game_user_2", "Command": "CU", "pw": "aaaaaaa", "email": "userTwo@email.com"}
[08/26/17 01:00:33 PM testhelper: 74] DEBUG Chn.name=Test2
[08/26/17 01:00:33 PM  router.py:102] DEBUG Parsing incoming json request: 
{"username": "game_user_2", "Command": "CU", "pw": "aaaaaaa", "email": "userTwo@email.com"}
[08/26/17 01:00:33 PM  router.py:142] DEBUG Received command: CU
[08/26/17 01:00:33 PM  router.py:189] DEBUG Adding channel name to JSON: Test22
[08/26/17 01:00:33 PM  router.py:247] DEBUG Response: {'Token': 'e05b07d92d0c4edf854f894550dc7848', 'Success': True}
[08/26/17 01:00:33 PM testhelper:120] DEBUG {"Token": "e05b07d92d0c4edf854f894550dc7848", "Success": true}
[08/26/17 01:00:33 PM testhelper: 74] DEBUG Chn.name=Test2
[08/26/17 01:00:33 PM  router.py:102] DEBUG Parsing incoming json request: 
{"username": "game_user_2", "Command": "LGN", "pw": "aaaaaaa"}
[08/26/17 01:00:33 PM  router.py:142] DEBUG Received command: LGN
[08/26/17 01:00:33 PM  router.py:189] DEBUG Adding channel name to JSON: Test22
[08/26/17 01:00:33 PM  router.py:247] DEBUG Response: {'Token': 'a45c5cd90bec4ada8debee6aca776998', 'Success': True}
[08/26/17 01:00:33 PM testhelper: 74] DEBUG Chn.name=Test2
[08/26/17 01:00:33 PM  router.py:102] DEBUG Parsing incoming json request: 
{"Ability": "Extra Range", "Perks": [], "Command": "ST", "Request_ID": "2", "Units": ["Cleric", "Archer", "Flier", "Armor", "Rogue", "Horseman", "Mage", "Swordsman"], "Leader": "Sniper"}
[08/26/17 01:00:33 PM  router.py:142] DEBUG Received command: ST
[08/26/17 01:00:33 PM  router.py:247] DEBUG Response: {'Success': True, 'Request_ID': u'2'}
[08/26/17 01:00:33 PM testhelper: 74] DEBUG Chn.name=Test2
[08/26/17 01:00:33 PM  router.py:102] DEBUG Parsing incoming json request: 
{"Command": "FM", "Request_ID": "3"}
[08/26/17 01:00:33 PM  router.py:142] DEBUG Received command: FM
[08/26/17 01:00:33 PM  router.py:247] DEBUG Response: {'Success': True, 'Request_ID': u'3'}
[08/26/17 01:00:33 PM   tasks.py: 50] DEBUG Processing matchmaking queue
[08/26/17 01:00:33 PM   tasks.py: 71]  INFO Found a match: game_user_2 vs. game_user_1
[08/26/17 01:00:33 PM   tasks.py: 83]  INFO Users have played each other 0 times.
[08/26/17 01:00:33 PM   tasks.py: 89]  INFO game_user_2 will take the first turn
[08/26/17 01:00:33 PM   tasks.py:101]  INFO Map chosen for this game: Map object
[08/26/17 01:00:33 PM   tasks.py:110]  INFO Created game id: 27
[08/26/17 01:00:33 PM   tasks.py:114] DEBUG Updating game_user entries
[08/26/17 01:00:33 PM   tasks.py:130] DEBUG Updating each player's unit entries
[08/26/17 01:00:33 PM   tasks.py:137] DEBUG Creating async notification messages
[08/26/17 01:00:33 PM   tasks.py:148] DEBUG Deleting players from the queue
[08/26/17 01:00:33 PM testhelper: 74] DEBUG Chn.name=Test
[08/26/17 01:00:33 PM  router.py:102] DEBUG Parsing incoming json request: 
{"Units": [{"Y": 0, "X": 0, "Name": "Archer"}, {"Y": 0, "X": 1, "Name": "Armor"}, {"Y": 0, "X": 2, "Name": "Cleric"}, {"Y": 0, "X": 3, "Name": "Flier"}, {"Y": 0, "X": 4, "Name": "Horseman"}, {"Y": 0, "X": 5, "Name": "Mage"}, {"Y": 0, "X": 6, "Name": "Rogue"}, {"Y": 0, "X": 7, "Name": "Swordsman"}], "Game": "vs. game_user_2 #1", "Command": "PU", "Request_ID": "4"}
[08/26/17 01:00:33 PM  router.py:142] DEBUG Received command: PU
[08/26/17 01:00:33 PM routehelpe: 75] ERROR Location X:0 Y:0 is not a valid placement location for a unit for your team.:
{u'Units': [{u'Y': 0, u'X': 0, u'Name': u'Archer'}, {u'Y': 0, u'X': 1, u'Name': u'Armor'}, {u'Y': 0, u'X': 2, u'Name': u'Cleric'}, {u'Y': 0, u'X': 3, u'Name': u'Flier'}, {u'Y': 0, u'X': 4, u'Name': u'Horseman'}, {u'Y': 0, u'X': 5, u'Name': u'Mage'}, {u'Y': 0, u'X': 6, u'Name': u'Rogue'}, {u'Y': 0, u'X': 7, u'Name': u'Swordsman'}], u'Game': u'vs. game_user_2 #1', 'session_username': u'game_user_1', u'Command': u'PU', u'Request_ID': u'4'}
[08/26/17 01:00:33 PM  router.py:247] DEBUG Response: {'Request_ID': u'4', 'Success': False, 'Error': 'Location X:0 Y:0 is not a valid placement location for a unit for your team.'}
```

--- a/Server/Game/tests.py
+++ b/Server/Game/tests.py
@@ -112,6 +112,34 @@ class TestGame(Communication.testhelper.CommonTestHelper):
 		# Ensure that there are two players in the queue at this point
 		self.assertTrue(Game_User.objects.filter(game=None).count() == 2)
 
+	def match_players_in_queue(self, success=True):
+		processMatchmakingQueue()
+
+		# Check if the matchmaking is expected to be successful
+		if not success:
+			return
+
+		game = Game.objects.latest('id')
+
+		game_users = Game_User.objects.filter(game=game)
+
+		# Ensure only 2 game users are in the game
+		self.assertEquals(len(game_users), 2)
+
+		# Ensure the first user created is always on team 1
+		first_game_user = game_users[0]
+		second_game_user = game_users[1]
+
+		if first_game_user.user == self.user:
+			first_game_user.team = 1
+			second_game_user.team = 2
+		else:
+			first_game_user.team = 2
+			second_game_user.team = 1
+
+		first_game_user.save()
+		second_game_user.save()
+
 class TestSetTeam(TestGame):
 	"""
 	All tests within this class relate specifically to the command 'ST' (Set team)
@@ -372,7 +400,7 @@ class TestMatchmaking(TestGame):
 		self.get_both_users_in_queue()
 
 	def test_mm_01_pair_success(self):
-		processMatchmakingQueue()
+		self.match_players_in_queue()
 
 		# Ensure the Game_User objects were created properly
 		self.assertTrue(len(Game_User.objects.filter(game=None)) == 0)
@@ -401,7 +429,7 @@ class TestMatchmaking(TestGame):
 		self.helper_execute_success(self.st_cmd)
 		self.helper_execute_success(self.fm_cmd)
 
-		processMatchmakingQueue()
+		self.match_players_in_queue(success=False)
 
 		# Ensure no game users were created
 		self.assertTrue(Game_User.objects.exclude(game=None).count() == 0)
@@ -410,14 +438,14 @@ class TestMatchmaking(TestGame):
 		self.assertTrue(Game_Queue.objects.count() == 1)
 
 	def test_mm_03_two_games_matched(self):
-		processMatchmakingQueue()
+		self.match_players_in_queue()
 
 		# Get both players in the queue again and match again
 		self.helper_execute_success(self.st_cmd)
 		self.helper_execute_success(self.fm_cmd)
 		self.helper_execute_success(self.st_cmd, 2)
 		self.helper_execute_success(self.fm_cmd, 2)
-		processMatchmakingQueue()
+		self.match_players_in_queue()
 
 		# Ensure the Game_User objects were created properly
 		self.assertTrue(len(Game_User.objects.filter(game=None)) == 0)
@@ -448,7 +476,7 @@ class TestMatchmaking(TestGame):
 		os.environ['UT-Fail'] = "True"
 
 		# Should throw an exception after the game is created in db
-		processMatchmakingQueue()
+		self.match_players_in_queue(success=False)
 
 		del os.environ['UT-Fail']
 
@@ -482,7 +510,7 @@ class TestPlaceUnits(TestGame):
 		super(TestPlaceUnits, self).setUp()
 
 		self.get_both_users_in_queue()
-		processMatchmakingQueue()
+		self.match_players_in_queue()
 		self.game = Game.objects.latest('pk')
 
 	def test_pu_01_bad_json(self):
@@ -600,12 +628,12 @@ class TestQueryGames(TestGame):
 
 		# setup game1
 		self.get_both_users_in_queue()
-		processMatchmakingQueue()
+		self.match_players_in_queue()
 		self.games.append(Game.objects.latest('pk'))
 
 		# setup game2
 		self.get_both_users_in_queue()
-		processMatchmakingQueue()
+		self.match_players_in_queue()
 		self.games.append(Game.objects.latest('pk'))
 
 		# QGU Command
@@ -988,7 +1016,7 @@ class TestTakeAction(TestGame):
 
 		# Get both users into a game
 		self.get_both_users_in_queue()
-		processMatchmakingQueue()
+		self.match_players_in_queue()
 		self.game = Game.objects.latest('pk')
 
 		# Place units for player one
@@ -2110,7 +2138,7 @@ class TestEndTurn(TestGame):
 
 		# Get both users into a game
 		self.get_both_users_in_queue()
-		processMatchmakingQueue()
+		self.match_players_in_queue()
 		self.game = Game.objects.latest('pk')
 
 		# Place units for player one

